integration/self_hosted/cluster.tf

```diff
@@ -0,0 +1,66 @@
+resource "materialize_cluster" "cluster" {
```

Comment:
The large number of line changes in this PR is due to all of the files in the `integration/self_hosted` directory. Those are just standard tests. If reviewing this PR, focus on the changes in the `pkg` directory.